packages/@aws-cdk/aws-cognito/lib/user-pool-idps/amazon.ts

```diff
@@ -0,0 +1,58 @@
+import { Construct, Resource } from '@aws-cdk/core';
+import { CfnUserPoolIdentityProvider } from '../cognito.generated';
+import { IUserPool } from '../user-pool';
+import { IUserPoolIdentityProvider } from '../user-pool-idp';
+
+/**
+ * Properties to initialize UserPoolAmazonIdentityProvider
+ */
+export interface UserPoolIdentityProviderAmazonProps {
+  /**
+   * The user pool to which this construct provides identities.
+   */
+  readonly userPool: IUserPool;
```

Comment:
Feels like this should be provided by a `bind()` method. Isn't it implied by the UserPoolClient you bind the identity provider to?

Reply:
Not sure I follow this here. An IdentityProvider is attached to a UserPool. 

The relationship between the IdentityProvider and UserPoolClient exists only to populate the `supportedIdentityProviders` CloudFormation property.

Reply:
Yeah, I was confused by the fact that the existing Identity Provider class BOTH created one for the User Pool AND returned its type for use with the Client.

The class was doing 2 things at the same time, which feels wrong.

Reply:
> BOTH created one for the User Pool AND returned its type for use with the Client.

Apologies, but I still don't understand the problem. Why is this a problem? Can you expand?

Reply:
IDP registration and IDP selection for use with a client should be 2 distinct things.

Potentially like so:

```
// Register IDP
userPool.registerIdentityProvider('Facebook', IdentityProvider.facebook({
  secret: 'xxxx',
});

new UserPoolClient(this, 'Client', {
  // Select IDP
  identityProviders: [IdentityProvider.FACEBOOK, IdentityProvider.COGNITO],
});
```

Reply:
It's sort of in reverse currently, not sure how important the distinction is.

```typescript
const idp = cognito.UserPoolIdentityProvider.facebook(this, 'FacebookIDP', {
            clientId: '...',
            clientSecret: secret.secretValue.toString(),
            scopes: ['email'],
            userPool
        });
```

Are we far enough along to merge this, since it's still experimental?